services/QAModelService.py

```diff
@@ -33,3 +34,11 @@ def read_latest_qa_model_by_type(self, model_type: string) -> QAModel:
 
         selected_model = QAModel.query.filter(QAModel.ml_type == model_type).order_by(QAModel.created_date.desc()).first()
         return selected_model
+
+    def read_latest_models(self) -> list:
+        '''Service function used to retrive the latest models for each type'''
+
+        #TODO: Not working correctly
+        models = QAModel.query.order_by(QAModel.created_date.desc()).distinct(QAModel.ml_type)
+        return models
```

Comment:
Not sure how custom SQL works in SQL Alchemy, but I think something similar to this should do the trick

```
select qa1.*
from qa_models qa1
where qa1.created_date in (
     select max(qa2.created_date)
     from qa_models qa2
     where qa2.ml_type = qa1.ml_type
)

Reply:
Will be visited  later added feedback to a comment in code with TODO